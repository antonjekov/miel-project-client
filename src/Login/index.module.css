
.form-control:focus {
    border-color: rgb(9, 211, 69);
    box-shadow: 0 0 0 0.2rem rgba(40, 167, 69, 0.25);
}

a{
    color: orange;
}

.Login{
    margin-top: 3em;
    margin-bottom: 3em;
    };



/* .Login{
     width: 83%; 
    display: inline-block;
    vertical-align: top;
} */

/* .form-control {
    width: 30%;
    margin: 0 auto;
    padding: 1%;
    text-align: center;
}

label {
    float: left;
    width: 30%;
    text-align: right;
}

input {
    text-align: center;
    border:1px solid #234465;
    padding: 1%;
    width: 50%;
    border-radius: 6px;
    color: #234465;
}

form button {
    width: auto;
    padding: 2%;
}

 */
