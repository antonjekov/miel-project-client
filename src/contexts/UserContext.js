import React from "react";

const UserContext = React.createContext({name: "Anton"});

export default UserContext