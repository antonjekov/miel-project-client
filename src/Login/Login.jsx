import { Form, Button, Spinner, Col } from 'react-bootstrap';
import styles from './Login.module.css';
import React, { useState } from 'react';
import userSchemaLogin from "../schemas/userSchemaLogin";
import { useFormik } from "formik";
import userService from "../services/user_service";
import { useAuth } from "../contexts/Auth";
import { useHistory } from "react-router-dom";

function Login(props) {
    const history = useHistory()
    const { setUserInfo } = useAuth();
    const [isLoading, setIsLoading] = useState(false);
    //Formik integration hook
    const { handleSubmit, handleChange, errors, values } = useFormik({
        initialValues: {
            username: '',
            password: ''
        },
        validationSchema: userSchemaLogin,
        onSubmit(values, actions) {
            setIsLoading(true);
            userService.login(values)
                .then(async res => {
                    if (!res.ok) {
                        setIsLoading(false);
                        const errorsObject = await res.json()
                        actions.setErrors(errorsObject)
                        return
                    }
                    setIsLoading(false);
                    const userInfo = await res.json();
                    setUserInfo(userInfo);
                    history.push('/');
                })
                .catch(err => {
                    console.log(err)//TO DO Must have global handler page for Server errors ...
                });
        }
    });

    return (
        <Col md={{ offset: 4, span: 4 }}>
            <div className={styles.Login}>
                <Form onSubmit={handleSubmit}>
                    <div>
                        <h1>Login</h1>
                    </div>
                    <br></br>
                    <Form.Row>
                        <Col>
                            <Form.Group >
                                <Form.Label>Email address</Form.Label>
                                <Form.Control className={styles['form-control']} value={values.username} onChange={handleChange} type="email" placeholder="Enter email" name='username' isInvalid={!!errors.username} isValid={values.username && !errors.username} />
                                <Form.Control.Feedback type='invalid' >{errors.username}</Form.Control.Feedback>
                                <Form.Control.Feedback type='valid'>Look's good</Form.Control.Feedback>
                            </Form.Group>
                        </Col>
                    </Form.Row>

                    <Form.Row>
                        <Col>
                            <Form.Group >
                                <Form.Label>Password</Form.Label>
                                <Form.Control className={styles['form-control']} value={values.password} onChange={handleChange} type="password" placeholder="Password" name='password' isInvalid={!!errors.password} isValid={values.password && !errors.password} />
                                <Form.Control.Feedback type='invalid' >{errors.password}</Form.Control.Feedback>
                                <Form.Control.Feedback type='valid'>Look's good</Form.Control.Feedback>
                            </Form.Group>
                        </Col>
                    </Form.Row>
                    {isLoading ?
                        <Spinner animation="border" variant="warning" /> :
                        <Button variant="warning" type="submit" >Login</Button>}
                    <Form.Text >
                        Don't have an account? <a href="/register" ><b>Register</b></a>
                    </Form.Text>
                </Form>
            </div>
        </Col>)
}

export default Login