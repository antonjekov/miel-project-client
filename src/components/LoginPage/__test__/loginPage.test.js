import React from "react";
import ReactDOM from "react-dom"
import {render, cleanup, screen, fireEvent} from "@testing-library/react"
import renderer from "react-test-renderer"
import Login from "../"
import { useAuth, AuthContext } from  "../../../contexts/Auth";

afterEach(cleanup)

const setUserInfo =()=>{}
test("inputs update on change", () =>{
    const {queryByPlaceholderText, queryByText} = render(<AuthContext.Provider value={{ setUserInfo}}><Login/></AuthContext.Provider>)
    const userInput = queryByPlaceholderText('Enter email')
    fireEvent.change(userInput,{target:{value: 'test'}})
    expect(userInput.value).toBe('test')
    const passwordInput = queryByPlaceholderText('Password')
    fireEvent.change(passwordInput,{target:{value: '123456'}})
    expect(passwordInput.value).toBe('123456')
  })