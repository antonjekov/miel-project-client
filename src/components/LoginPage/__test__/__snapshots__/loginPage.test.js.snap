// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="col-md-4 offset-md-4"
>
  <div
    className="FormWrapper"
  >
    <h1>
      Login
    </h1>
    <form
      className=""
      onSubmit={[Function]}
    >
      <div
        className="form-row"
      >
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Email address
            </label>
            <input
              className="form-control form-control"
              name="username"
              onChange={[Function]}
              placeholder="Enter email"
              type="email"
              value=""
            />
            <div
              className="invalid-feedback"
            />
            <div
              className="valid-feedback"
            >
              Look's good
            </div>
          </div>
        </div>
      </div>
      <div
        className="form-row"
      >
        <div
          className="col"
        >
          <div
            className="form-group"
          >
            <label
              className="form-label"
            >
              Password
            </label>
            <input
              className="form-control form-control"
              name="password"
              onChange={[Function]}
              placeholder="Enter your password"
              type="password"
              value=""
            />
            <div
              className="invalid-feedback"
            />
            <div
              className="valid-feedback"
            >
              Look's good
            </div>
          </div>
        </div>
      </div>
      <button
        className="btn btn-warning"
        disabled={false}
        type="submit"
      >
        Login
      </button>
      <small
        className="form-text"
      >
        Don't have an account? 
        <a
          href="/register"
        >
          <b>
            Register
          </b>
        </a>
      </small>
    </form>
  </div>
</div>
`;
